tests/bugs/gh-4971.slang

```diff
@@ -0,0 +1,25 @@
+//TEST(compute):COMPARE_COMPUTE(filecheck-buffer=CHECK):-vk -output-using-type
+//TEST(compute):COMPARE_COMPUTE(filecheck-buffer=CHECK):-d3d11 -output-using-type
+
+struct Test {
+    RWStructuredBuffer<int> val;
+    __subscript(int x, int y)->int
+    {
+        get { return val[x * 3 + y]; }
+        [nonmutating] set { val[x * 3 + y] = newValue; }
+    }
+}
+Test test;
```

Comment:
I think two tests can be combined,
```
//TEST:SIMPLE(filecheck=CHECK): -target spirv -DERROR_CASE
//TEST(compute):COMPARE_COMPUTE(filecheck-buffer=CHECK):-vk -output-using-type
//TEST(compute):COMPARE_COMPUTE(filecheck-buffer=CHECK):-d3d11 -output-using-type

struct Test {
    RWStructuredBuffer<int> val;
    __subscript(int x, int y)->int
    {
        get { return val[x * 3 + y]; }
#if !defined(ERROR_CASE)
        [nonmutating]
#endif
        set { val[x * 3 + y] = newValue; }
    }
}
Test test;
```

Reply:
I think both ways are fine, personally it is just cleaner to me to keep things separate.